src/applications/personalization/profile360/actions/updaters.js

```diff
@@ -37,17 +39,37 @@ function recordProfileTransaction(event, fieldName) {
 }
 
 export function refreshTransaction(transaction, analyticsSectionName) {
-  return async (dispatch) => {
+  return async (dispatch, getState) => {
     try {
       const { transactionId } = transaction.data.attributes;
-      const transactionRefreshed = isVet360Configured() ? await apiRequest(`/profile/status/${transactionId}`) : localVet360.updateTransactionRandom(transactionId);
+      const state = getState();
+      const isAlreadyAwaitingUpdate = state.vet360.transactionsAwaitingUpdate.includes(transactionId);
+
+      if (isAlreadyAwaitingUpdate) {
+        return;
+      }
+
```

Comment:
The transaction component will call this function every second. However, the function exits early if there is still a pending request, to keep transactions from piling up in case we're polling more quickly than the API can respond.